### PR TITLE
Add load error capturing for callback.

### LINKED ISSUE #15: Feature: Add `script.onerror` handler.
* [x] Add `onerror` handler
* [x] Test errors down to IE8 (may be future work).
* [x] Figure out IE8 `script.onerror` story.
* [x] Update doc byte sizes.
* [x] Update doc usage stuff.

New API:

```js
window._lload("http://example.com/foo.js", function (err) {
  if (err) { console.log("OH NOES", err); }
  // foo.js is loaded!
}/*, [optional context variable here] */);
```

/cc @PeoB

### PR DESCRIPTION
**Pull request by ryan-roemer**
Holy moly, it works!!!

Fixes #15

* Use script.onerror for IE9+
* Use horrible hack for IE8
* Add unit and functional tests
* Update documentation and byte size

/cc @exogen @PeoB @aisapatino @baer

### CODE REVIEW

**Comment by ryan-roemer** [comment]
Travis hasn't picked up the PR yet. A SO comment says "close and reopen" can force a build, so doing that.

http://stackoverflow.com/questions/19589260/force-travis-to-build-a-specific-pull-request-or-commit-sha-that-it-missed

**Comment by ryan-roemer** [comment]
Weird, that worked. Travis is a 'runnin.

File: README.md [hunk 1]
```diff
@@ -39,9 +41,9 @@ If you use an AMD bundling tool (like RequireJS):
 
 ```js
 define(["little-loader"], function (load) {
-  load("http://example.com/foo.js", function () {
-    // foo.js is loaded!
-  }/*, [optional context (`this`) variable here] */);
+  load("http://example.com/foo.js", function (err) {
+    // ... your code ...
+  });
```
Comment (baer) [inline README.md]: No more context?

**Comment by baer** [comment]
Wowza - nice work!

File: lib/little-loader.js [hunk 2]
```diff
@@ -56,11 +57,50 @@
...
+
+    /**
+     * Create handler for error or completion.
+     *
+     * **Note**: Will only be called _once_.
+     *
+     * @param {Function}  finishCb Internal state cleanup.
+     * @returns {Function}         Callback wrapper
+     */
+    var _createFinish = function (finishCb) {
```
Comment (PeoB) [inline lib/little-loader.js]: Could call it _once to make intent clearer

File: README.md (hunk 1 above)
Comment (ryan-roemer) [inline README.md]: @baer -- It's still there and in the first example, just trying to cut down on repeated big comment pastes here and below.

Comment (baer) [inline README.md]: Yup, that's fine. Carry on

**Comment by exogen** [comment]
Going over this in more detail. In general though I'd be *very* cautious about making such big changes without having a good test case first. I'm almost certain that most of the incorrect loaders would pass our current test suite here, and there none of the tests yet exercise the failure scenario. So even if we're *pretty* sure that the non-error-case logic is kept the same, it would be nice to know for sure!

**Comment by ryan-roemer** [comment]
@exogen -- Teams need to be able to tell if a script fail right now. They're using `onerror` unreliably in VanillaJS code.

Can you open up a PR with a functional test to expose the issue or paste a diff here?

**Comment by exogen** [comment]
To follow up: this branch still passes the more brutal test I have in my repo, so it doesn't break anything, that's good! It sounds like @ryan-roemer is working on porting that test, as I'm mostly down for the count today. Still testing the new `onerror` handling.

**Pull request merged**